### PR TITLE
Get rid of deprecated components

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Removes:
- mentions of the Node.js bindings
- mentions of the Scala bindings
- usage of the Scala codegen as an SDK tool
- Java bindings "reactive components"

changelog_begin
The Node.js bindings are no longer supported as part of the Daml SDK
The Scala bindings are no longer supported as part of the Daml SDK
The Java bindings reactive components are no longer supported as part of the Daml SDK
The `daml codegen` command has dropped support for Scala
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/app-dev/bindings-x-lang/index.rst [hunk 1]
```diff
@@ -22,12 +22,10 @@ Note that for different reasons we chose codegen, but that is not the only optio
 
 - `Template Haskell <https://wiki.haskell.org/Template_Haskell>`_
 
-- Scala macro annotations (not future-proof enough to use when implementing the last Scala codegen)
-
 Building Ledger Commands
 ========================
 
-No matter what approach you take, either manually building commands or writing a codegen to do this, you need to understand how ledger commands are structured. This section demonstrates how to build create and exercise commands manually and how it can be done using contract classes generated by Scala codegen.
+No matter what approach you take, either manually building commands or writing a codegen to do this, you need to understand how ledger commands are structured. This section demonstrates how to build create and exercise commands manually and how it can be done using contract classe.
```
Comment (cocreature) [inline docs/source/app-dev/bindings-x-lang/index.rst:28]:
```suggestion
No matter what approach you take, either manually building commands or writing a codegen to do this, you need to understand how ledger commands are structured. This section demonstrates how to build create and exercise commands manually and how it can be done using contract classes.
```

**Comment by stefanobaghino-da** [comment]
5y3igt

**Comment by cocreature** [comment]
I want to see non-deprecated components burn as well

**Review by robin-da — APPROVED** [review]
Looks great, thanks!

File: language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala [hunk 2]
```diff
@@ -140,7 +139,6 @@ object CodegenConfigReader {
 
   private def dest(a: CodegenDest): String = a match {
     case Java => "java"
-    case Scala => "scala"
```
Comment (cocreature) [inline language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala:143]: this isn’t going to fly. Canton currently consumes this JAR to run the Scala codegen. We either need to release a standalone JAR using just the Scala codegen or leave this for now.

**Comment by robin-da** [comment]
Should this be :fire: as well: https://github.com/digital-asset/daml/blob/main/daml-assistant/daml-helper/src/DA/Daml/Helper/Main.hs#L244 ?

File: ledger-api/README.md [hunk 3]
```diff
@@ -2,15 +2,11 @@
 
 This is the API code for the ledger, which contains:
  * gRPC API definitions
- * Generated Scala bindings
+ * Generated Scala bindings and client
```
Comment (akshayshirahatti-da) [inline ledger-api/README.md:5]: I am probably being daft , but what is the reason for this change ? thought we are removing all mentions of it

Comment (stefanobaghino-da) [reply ledger-api/README.md:5]: This is a README of an internal directory which lists the directory's content.

File: docs/source/tools/codegen.rst [hunk 4]
```diff
@@ -83,29 +83,20 @@ Here is an example::
...
-      scala:
-        package-prefix: com.daml.quickstart.iou
-        output-directory: scala-codegen/src/main/scala
-        verbosity: 2
 
-You can then run the above configuration to generate your **Java** or **Scala** code::
+You can then run the above configuration to generate your **Java** code::
 
-    $ daml codegen [js|java|scala]
+    $ daml codegen java
```
Comment (akshayshirahatti-da) [inline docs/source/tools/codegen.rst:89]: again probably a silly one , should we get be keeping `js` here given the changes above ? and the line below ?

**Review by akshayshirahatti-da — APPROVED** [review]
LGTM , couple of minor comments

File: language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala:143]: Thanks for making me aware of this. I'll add the standalone JAR as part of this PR. Would you mind blocking it to avoid accidental merges? I already have a couple of approvals and I can just dismiss your block once I did the work to have the standalone JAR. 🙇🏻

File: docs/source/tools/codegen.rst (hunk 4 above)
Comment (stefanobaghino-da) [reply docs/source/tools/codegen.rst:89]: The line above says:

> You can then run the above configuration to generate your **Java** code::

So I assumed that the inclusion of `js` was a mistake. @robin-da?

**Comment by stefanobaghino-da** [comment]
> Should this be fire as well: https://github.com/digital-asset/daml/blob/main/daml-assistant/daml-helper/src/DA/Daml/Helper/Main.hs#L244 ?

I'll remove it, thanks for making me notice that. 🙇🏻

**Review by cocreature — CHANGES_REQUESTED** [review]
requesting changes until we publish a standalone JAR for the scala codegen.

**Comment by stefanobaghino-da** [comment]
> Should this be fire as well: https://github.com/digital-asset/daml/blob/main/daml-assistant/daml-helper/src/DA/Daml/Helper/Main.hs#L244 ?

Should be addressed by https://github.com/digital-asset/daml/pull/12167/commits/0b44000dbb7bb045b7d25ebe7f30501b641cf08e, let me know if it looks in order. 🙇🏻

File: docs/source/tools/codegen.rst (hunk 4 above)
Comment (robin-da) [reply docs/source/tools/codegen.rst:89]: I also think this is a mistake.

**Review by cocreature — APPROVED** [review]

**Comment by stefanobaghino-da** [comment]
@oliverse-da FYI from the next snapshot (which I believe won't be cut before next year) you should use the standalone Scala codegen JAR instead of `daml codegen scala`.

File: language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala (hunk 2 above)
Comment (stefanobaghino-da) [reply language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala:143]: Addressed by https://github.com/digital-asset/daml/pull/12167/commits/8a03f8faa108fb416176d941aee0d1b41b28f8d8

**Comment by oliverse-da** [comment]
> @oliverse-da FYI from the next snapshot (which I believe won't be cut before next year) you should use the standalone Scala codegen JAR instead of `daml codegen scala`.

Thanks for the heads-up @stefanobaghino-da - this means we will use the `codegen-jvm-main*.jar` instead, right? I.e. based on what I have prototyped and tested successfully here: https://github.com/DACH-NY/canton/pull/8114/files#diff-486adab54dc0e9a504950abc7f803ca7833458c1883e114101905a992480a782R426-R431

**Comment by cocreature** [comment]
Not quite, you’ll have to adapt to `com.daml.codegen-scala-main`. I think the only difference is that it does not include the Java codegen and doesn’t have the `scala` parameter as the first positional argument.
